app/scripts/controllers/swaps.types.ts

```diff
@@ -121,6 +323,45 @@ export type QuoteRequest = {
   walletAddress: string;
 };
 
+export type SwapsControllerOptions = {
+  getBufferedGasLimit: (
+    params: {
+      txParams: {
+        value: string;
+        data: string;
+        to: string;
+        from: string;
+      };
+    },
+    factor: number,
+  ) => Promise<{ gasLimit: string; simulationFails: boolean }>;
+  provider: ExternalProvider | JsonRpcFetchFunc;
+  getProviderConfig: () => ProviderConfig;
+  getTokenRatesState: () => {
+    marketData: Record<
+      string,
+      {
+        [tokenAddress: string]: {
+          price: number;
+        };
+      }
+    >;
+  };
```

Comment:
This can now be achieved using the `TokenRatesController:getState` action. We should replace constructor option callbacks with messenger actions/events wherever possible as part of the BaseControllerV2 migration.

Reply:
Updated!